src/gui/src/components/dashboard-grid/dasboard-table.tsx

```diff
@@ -0,0 +1,117 @@
+import { DataTable } from '@/components/data-table/data-table.jsx'
+import { dashboardColumns } from '@/components/dashboard-grid/table-columns.jsx'
+import { DEFAULT_QUERY, useGraphStore } from '@/state/index.js'
+import {
+  type Action,
+  type DashboardDataProject,
+} from '@/state/types.js'
+import {
+  type VisibilityState,
+  type Table,
+} from '@tanstack/react-table'
+
+interface DashboardTableProps {
+  data: DashboardDataProject[]
+  setTable: React.Dispatch<
+    React.SetStateAction<Table<DashboardDataProject> | null>
+  >
+  tableFilterValue: string
+  columnVisibility: VisibilityState
+  setColumnVisibility: React.Dispatch<
+    React.SetStateAction<VisibilityState>
+  >
+}
+
+type SelectDashboardItemOptions = {
+  updateActiveRoute: Action['updateActiveRoute']
+  updateErrorCause: Action['updateErrorCause']
+  updateQuery: Action['updateQuery']
+  updateStamp: Action['updateStamp']
+  item: DashboardDataProject
+}
+
+const selectDashboardItem = async ({
+  updateActiveRoute,
+  updateErrorCause,
+  updateQuery,
+  updateStamp,
+  item,
+}: SelectDashboardItemOptions) => {
+  let req
+  try {
+    req = await fetch('/select-project', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        path: item.path,
+      }),
+    })
+  } catch (err) {
+    console.error(err)
+    updateActiveRoute('/error')
+    updateErrorCause('Failed to request project selection.')
+    return
+  }
+
+  let projectSelected = false
+  try {
+    projectSelected = (await req.json()) === 'ok'
+  } catch (err) {
+    console.error(err)
+  }
+
+  if (projectSelected) {
+    window.scrollTo(0, 0)
+    updateQuery(DEFAULT_QUERY)
+    updateActiveRoute('/explore')
+    updateStamp()
+  } else {
+    updateActiveRoute('/error')
+    updateErrorCause('Failed to select project.')
+  }
+}
+
+const DashboardTable = ({
```

Comment:
```suggestion
export const DashboardTable = ({
```

I think the standard has been to export each component function directly, is there any particular reason for changing it to exporting an object ref at the end?

Reply:
if not, let's make sure to change all files to export the function instead 🙏 i noticed most components from this PR are like this